src/Param.cc

```diff
@@ -675,6 +713,11 @@ bool ParamPrivate::ValueFromStringImpl(const std::string &_typeName,
         return ParsePoseUsingStringStream(
             tmp, this->key, {}, _valueToSet);
       }
+      else
```

Comment:
I'm guessing that this is to handle `<pose rotation_format="quat_xyzw" />`. The default string is a 6-tuple of zeros, while a `quat_xyzw` wants a 7-tuple. I believe this logic parses without attributes if the element value string is empty to sidestep the default value not working for `quat_xyzw`. If that is the case, I would prefer being more explicit about that in the comments for this logic.

Reply:
@scpeters is correct, that was one of the issues. Another issue that I found was this test, where the default value was not set, https://github.com/ignitionrobotics/sdformat/blob/sdf11/src/Utils_TEST.cc#L55-L68.

After looking at all the other breakages, this particular line in `Utils_TEST` seem to be an outlier that I think should be fixed, instead of accommodating.

Reply:
I've moved the default value logic into `ParsePoseUsingStringStream` and removed the check for `tmp.empty()`, let me know what you think, https://github.com/ignitionrobotics/sdformat/pull/690/commits/d78976a4e212b1ea2b7b864f82710241f77a8fb1

Reply:
Yes, I think you're right about that test. It was testing that empty poses are interpreted as identity per http://sdformat.org/tutorials?tut=pose_frame_semantics&cat=specification&#pose-and-frame, but yeah, it should use the default value as the means to do that instead of treating `//pose` as a special tag.

With the changes you made, `<pose rotation_format="quat_xyzw" />` doesn't seem to work anymore. Was that intentional?